src/tests/api_compat.rs

```diff
@@ -0,0 +1,12 @@
+//! This file is intended to be used as a stupidity-check when it comes
+//! to API compatibility. If you've broken something here, you've broken
+//! the public API.
+
+// TODO: This should be completed with all methods from the `Theme` trait.
+
+use crate::{theme::ClackTheme, Theme};
+
+#[test]
+fn format_note() {
+    ClackTheme.format_note("my prompt", "my message");
+}
```

Comment:
Just add it to the bottom of `src/theme.rs` (it's more Rust idiomatic: tests should be closer to the body).

```rust
....

#[test]
mod test {
    use super::*;

    #[test]
    fn format_note() {
          ClackTehem.format_note("backward compatibility", "check");
    }
}
```

Reply:
Updated!

I do in general disagree regarding idiomacy. it's a common pattern of course, but often distracts from business logic, and most recently I've seen a change towards such a separation. But I'm relatively new to :crab: , so.. that's just my opinion :smile:

Reply:
> @cylewitruk The best thing is the isolation of tests to the particular module and easiness of finding, editing, and removing all together in a single place.

I guess my aversion is largely related to working in an already-legacy codebase where 10k LoC files and the majority is tests is not uncommon :sweat_smile:  But I'm all for this in smaller modules -- makes it easier with direct access to privates as well.